// discount_service.proto
syntax = "proto3";

service DiscountService {
  rpc GenerateCodes (GenerateRequest) returns (GenerateResponse) {}
  rpc UseCode (UseCodeRequest) returns (UseCodeResponse) {}
}

message GenerateRequest {
  uint32 count = 1;
  uint32 length = 2;
}

message GenerateResponse {
  bool result = 1;
}

message UseCodeRequest {
  string code = 1;
}

message UseCodeResponse {
  uint32 result = 1;
}